pkgs/applications/networking/instant-messengers/teams/default.nix

```
@@ -0,0 +1,124 @@
+{ lib
+, stdenv
+, fetchurl
+, autoPatchelfHook
+, wrapGAppsHook
+, dpkg
+, libuuid
+, nspr
+, nss
+, alsaLib
+, pulseaudio
+, cups
+, expat
+, dbus
+, glib
+, cairo
+, pango
+, atk
+, at-spi2-atk
+, gtk3
+, gdk-pixbuf
+, libxcb
+, libxkbfile
+, libX11
+, libXext
+, libXfixes
+, libXi
+, libXcursor
+, libXcomposite
+, libXdamage
+, libXrender
+, libXrandr
+, libXtst
+, libXScrnSaver
+, libsecret
+, libgnome-keyring
+, coreutils
+, gawk
+, xdg_utils
+, systemd }:
+
+assert stdenv.hostPlatform.system == "x86_64-linux";
+
+let
+
+  version = "1.2.00.32451";
+
+in
+
+stdenv.mkDerivation {
```

```suggestion
stdenv.mkDerivation rec {
```

this allows self-referencing in the set, so you can move `version` into it and not inherit it.